src/index.ts

```diff
@@ -135,14 +137,15 @@ export function resolve(
   found: boolean
   path?: string | null
 } {
-  if (!cachedOptions || cachedOptions !== options) {
+  if (!cachedOptions || previousOptions !== options) {
+    previousOptions = options
     cachedOptions = {
       ...options,
       conditionNames: options?.conditionNames ?? defaultConditionNames,
       extensions: options?.extensions ?? defaultExtensions,
       extensionAlias: options?.extensionAlias ?? defaultExtensionAlias,
       mainFields: options?.mainFields ?? defaultMainFields,
-      fileSystem,
+      fileSystem: new CachedInputFileSystem(fileSystem, 4000),
```

Comment:
Why `4000` is chosen here?

Reply:
That's the number of milliseconds to cache for. I just copied it from https://github.com/webpack/enhanced-resolve#creating-a-resolver

I don't know what a good value for that would be if ever.

Reply:
I'd like to keep it as minimal as possible, can you help to try is `500` resulting same?

Reply:
In the provided test case, it's still 9 seconds so still good. But I tried in our production code base and it does get slower with 500.

Reply:
> In the provided test case, it's still 9 seconds so still good. But I tried in our production code base and it does get slower with 500.

OK, then I'm fine to increase it to 5000.